### PR TITLE
refactor(server): Update ws to address CVE

### PR DESCRIPTION
**Pull request by alexvy86**
## Description

Updates dependencies to get to ws@8.17.1 (or ws@7.5.10) to address https://nvd.nist.gov/vuln/detail/CVE-2024-37890. Updating socket.io to 4.8.0 was necessary in some cases get the necessary dependency ranges.

socket.io 4.7.5-4.8.0 is a minor semver update but contains [a breaking change in the type of the `close()` function](https://github.com/socketio/socket.io/pull/4971/files), so two places had to be updated to account for that.

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

[AB#8369](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/8369)

### CODE REVIEW

File: server/routerlicious/packages/services-shared/src/socketIoServer.ts [hunk 1]
```diff
@@ -321,8 +321,12 @@ class SocketIoServer implements core.IWebSocketServer {
 			}
 		}
 
-		this.io.close();
-		await sleep(3000); // Give time for any  disconnect handlers to execute before closing Redis resources
+		await this.io.close();
```
Comment (alexvy86) [inline server/routerlicious/packages/services-shared/src/socketIoServer.ts:324]: Necessary due to breaking change in socket.io `4.7.5->4.8.0`.

File: server/routerlicious/packages/tinylicious/src/services/webServerFactory.ts [hunk 2]
```diff
@@ -71,7 +71,7 @@ class SocketIoServer extends EventEmitter implements IWebSocketServer {
 	}
 
 	public async close(): Promise<void> {
-		await new Promise<void>((resolve) => this.io.close(() => resolve()));
+		await this.io.close();
```
Comment (alexvy86) [inline server/routerlicious/packages/tinylicious/src/services/webServerFactory.ts:74]: Necessary due to breaking change in socket.io `4.7.5->4.8.0`.

File: server/routerlicious/packages/services-shared/src/socketIoServer.ts (hunk 1 above)
Comment (alexvy86) [reply server/routerlicious/packages/services-shared/src/socketIoServer.ts:324]: @znewton any thoughts on this change?

File: server/routerlicious/packages/tinylicious/src/services/webServerFactory.ts (hunk 2 above)
Comment (noencke) [reply server/routerlicious/packages/tinylicious/src/services/webServerFactory.ts:74]: Can we get even shorter? If `this.io.close()` returns `Promise<void>`, then this function doesn't even need to be async anymore, correct? You can just do:

```ts
public close(): Promise<void> {
  return this.io.close();
}
```

Comment (alexvy86) [reply server/routerlicious/packages/tinylicious/src/services/webServerFactory.ts]: Yep, updated.

**Review by Josmithr — APPROVED** [review]

File: server/routerlicious/packages/services-shared/src/socketIoServer.ts (hunk 1 above)
Comment (alexvy86) [reply server/routerlicious/packages/services-shared/src/socketIoServer.ts:324]: I feel decently confident that this shouldn't have a big impact so I'll go ahead merging.

**Pull request merged**